### PR TITLE
Add extensions to support draft 3D API

### PR DESCRIPTION
**Pull request by vincentmarchetti**
This will bring development of the manifesto code to support the draft 3d manifests prepared by IIIF 3D Technical study group. This branch will support developing and testing the manifesto code against the draft 3D manifests in [IIIF/3d](https://github.com/IIIF/3d) . By including compiled manifesto modules it also supports installation by npm into test viewer code.

### CODE REVIEW

File: src/RotateTransform.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    
+    getRotation() : object {
+        var retVal = {};
+        for (const attrib of ["x","y","z"]){
+            var raw = this.__jsonld[attrib];
+            retVal[attrib] = (raw !== undefined)?Number(raw):0.0;
+        }
+        return retVal;
```
Comment (stephenwf) [inline src/RotateTransform.ts:19]: This is quite a confusing way to write this.
```ts
return {
  x: this.__jsonld.x ? parseFloat(this.__jsonld.x) : 0.0,
  y: this.__jsonld.y ? parseFloat(this.__jsonld.y) : 0.0,
  z: this.__jsonld.z ? parseFloat(this.__jsonld.z) : 0.0,
};
```

Also, should the rotation also be using the Vec3?

File: src/Scene.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    return content;
+  };
+  
+  // 3D extension
+  get Content() : Annotation[] { return this.getContent(); }
+  
+  getAnnotationById( searchId: string ) : Annotation | null {
+    for (var anno of this.Content )
+        if (anno.id === searchId ) return anno;
+    return null;
```
Comment (stephenwf) [inline src/Scene.ts:52]: Will this create a new instance of `AnnotationPage()` each time you request an annotation by ID? Could the annotation page be created in the constructor from the jsonld property?

File: .gitignore [hunk 3]
```diff
@@ -1,9 +1 @@
-dist-commonjs
-dist-esmodule
-dist-umd
-dist-var
-coverage
-npm-debug.log
 node_modules
-debug.log
-types
```
[the remark this answers is not in the document]
Comment (vincentmarchetti) [reply .gitignore:9]: Thanks for the suggestion on the ci.json

I cleared out .gitignore as part of my refactoring to prepare this for the pull request, and neglected to restore it. I will do so using the Manifold example

**Comment by stephenwf** [comment]
Not sure what the HTTP status codes issue is. Maybe @edsilv ?

File: src/RotateTransform.ts (hunk 1 above)
Comment (vincentmarchetti) [reply src/RotateTransform.ts:19]: I agree on the cleaner version of the code for extracting x,y,z values.

Practically, the suggestion for using Vec3 as the return value would work.

Mathematically, the values for RotateTransform and ScaleTransform are just 3 numbers labelled x,y,z; geometrically not notvectors. My preference is to not use a vector class in this case. I have already lost this argument in the context of the ISO X3D spec, so I would not really object to using Vec3 in this case as well :)

**Comment by vincentmarchetti** [comment]
On seeing the failed ci executions for Node 14 I suspected its an 'old Node version' problem. On codesandbox it looked like it was succeeding for Node 16, but I'm not confident enough with any of the tools to say for sure. I saw that ci.json specified "node" : "14"

**Pull request merged**